packages/jest-haste-map/src/index.js

```diff
@@ -325,6 +325,28 @@ class HasteMap extends EventEmitter {
           throw new Error(message);
         }
         this._console.warn(message);
+        // We do NOT want consumers to use a module that is ambiguous.
+        delete moduleMap[platform];
+        if (Object.keys(moduleMap).length === 1) {
+          delete map[id];
+        }
+        let dupsByPl = hasteMap.duplicates[id];
+        if (dupsByPl == null) {
+          dupsByPl = hasteMap.duplicates[id] = (Object.create(null): any);
```

Comment:
these `any` are necessary, otherwise Flow complains below that is cannot find "g" in the object. The problem is that we're precisely assigning "g", not trying to fetch it, so that's a little weird. I think it's a false positive from Flow.